crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -78,50 +74,82 @@ pub struct Mesh {
-/// - It is normal for multiple vertices to have the same position
-///   attribute - it's a common technique in 3D modelling for complex UV mapping or other calculations.
 ///
-/// To render correctly with `StandardMaterial` a mesh needs to have properly defined:
-/// - [`UVs`](Mesh::ATTRIBUTE_UV_0): Bevy needs to know how to map a texture onto the mesh.
-/// - [`Normals`](Mesh::ATTRIBUTE_NORMAL): Bevy needs to know how light interacts with your mesh. ([0.0, 0.0, 1.0] is very
-///              common for simple meshes because simple meshes are smooth, and they don't require complex light calculations.)
-/// - Vertex winding order -
-///              the default behavior is with `StandardMaterial.cull_mode` = Some([`Face::Front`](crate::render_resource::Face::Front)) which means
-///              that by default Bevy would *only* render the front of each triangle, and the front
-///              is the side of the triangle in which the vertices appear in a *counter-clockwise* order.
+/// You can see how it looks like [here](https://github.com/bevyengine/bevy/blob/main/assets/dovs/Mesh.png),
+/// used in a `PbrBundle` with a square bevy logo texture, with added axis, points,
+/// lines and text for clarity.
+///
+/// For further visualization, explanation, and examples see the built-in Bevy examples,
+/// in particular [generate_custom_mesh](https://github.com/bevyengine/bevy/blob/main/examples/3d/generate_custom_mesh.rs),
+/// and the [implementation of the built-in shapes](https://github.com/bevyengine/bevy/tree/main/crates/bevy_render/src/mesh/shape).
+///
+/// ## Common points of confusion
 ///
+/// - UV maps in Bevy are "flipped", `[0.0, 0.0]` = Top-Left (not Bot-Left like `OpenGL`).
+/// - It is normal for multiple vertices to have the same [position attribute](Mesh::ATTRIBUTE_POSITION) value,
+/// it's a common technique in 3D modelling for complex UV mapping or other calculations.
+///
+/// ## Use with `StandardMaterial`
+///
+/// To render correctly with `StandardMaterial`, a mesh needs to have properly defined:
+/// - [`UVs`](Mesh::ATTRIBUTE_UV_0): Bevy needs to know how to map a texture onto the mesh
+/// (also true for `ColorMaterial`).
+/// - [`Normals`](Mesh::ATTRIBUTE_NORMAL): Bevy needs to know how light interacts with your mesh.
+/// [0.0, 0.0, 1.0] is very common for simple flat meshes on the XY plane,
+/// because simple meshes are smooth and they don't require complex light calculations.
+/// - Vertex winding order: by default, `StandardMaterial.cull_mode` is [`Some(Face::Front)`](crate::render_resource::Face),
+/// which means that Bevy would *only* render the "front" of each triangle, which
+/// is the side of the triangle from where the vertices appear in a *counter-clockwise* order.
+///
+// TODO: allow values to be unloaded after been submitting to the GPU to conserve memory
+#[derive(Debug, TypeUuid, TypePath, Clone)]
+#[uuid = "8ecbac0f-f545-4473-ad43-e1f4243af51e"]
+pub struct Mesh {
```

Comment:
Since you're already improving the docs. It might be worth it to also add short example of actually getting back an attribute from the mesh. The code to do that can be a little convoluted so having a small code block showing it could be useful.

Reply:
This is covered in the `generate_custom_mesh` example that I forgot to link in the PR description: https://github.com/bevyengine/bevy/pull/8909

But I did link to it in this PR code, under the doc example.

I don't think it's not worth making a really big doc example, but I can clarify that `generate_custom_mesh` does that!

Reply:
Ah, good to know, I didn't know that was in that example